test/abstract_client.js

```diff
@@ -1536,5 +1536,52 @@ module.exports = function (server, config) {
         }
       })
     })
+
+    context('with alternate server client', function () {
+      var cachedClientListeners
+
+      beforeEach(function () {
+        cachedClientListeners = server.listeners('client')
+        server.removeAllListeners('client')
+      })
+
+      afterEach(function () {
+        server.removeAllListeners('client')
+        cachedClientListeners.forEach(function (listener) {
+          server.on('client', listener)
+        })
+      })
+
+      it('should resubscribe even if disconnect is before suback', function (done) {
+        var client = mqtt.connect(Object.assign({ reconnectPeriod: 100 }, config))
+        var subscribeCount = 0
+        var connectCount = 0
+
+        server.on('client', function (serverClient) {
```

Comment:
Yeah. It doesn't work. 

First of all it has to be done twice since the client reconnects, and even if I code around that, the existing event listeners blow up when I sever the connection in the initial subscription. TBH, I'm a bit surprised the rest of the tests work as well as they do that way.